atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1451,19 +1492,132 @@ private static SnapshotTransaction unwrapSnapshotTransaction(Transaction transac
+                MetricsManager metricsManager,
+                KeyValueService keyValueService,
+                TimestampService timestampService,
+                TimestampManagementService timestampManagementService,
+                LockClient lockClient,
+                LockService lockService,
+                TransactionService transactionService,
+                ConflictDetectionManager conflictDetectionManager,
+                SweepStrategyManager sweepStrategyManager,
+                MultiTableSweepQueueWriter sweepQueue,
+                ExecutorService deleteExecutor,
+                BiFunction<Transaction, AtomicInteger, Transaction> transactionWrapper,
+                BiFunction<KeyValueService, AtomicInteger, KeyValueService> keyValueServiceWrapper) {
+            super(
+                    metricsManager,
+                    keyValueService,
+                    timestampService,
+                    timestampManagementService,
+                    lockClient,
+                    lockService,
+                    transactionService,
+                    conflictDetectionManager,
+                    sweepStrategyManager,
+                    sweepQueue,
+                    deleteExecutor);
             this.transactionWrapper = transactionWrapper;
+            this.keyValueServiceWrapper = keyValueServiceWrapper;
+        }
+
+        @Override
+        public Transaction createNewTransaction() {
+            AtomicInteger counter = new AtomicInteger();
+            long startTimestamp = timelockService.getFreshTimestamp();
+            return transactionWrapper.apply(
+                    new SnapshotTransaction(
+                            metricsManager,
+                            keyValueServiceWrapper.apply(keyValueService, counter),
+                            timelockService,
+                            transactionService,
+                            NoOpCleaner.INSTANCE,
+                            () -> startTimestamp,
+                            getConflictDetectionManager(),
+                            SweepStrategyManagers.createDefault(keyValueService),
+                            startTimestamp,
+                            Optional.empty(),
+                            PreCommitConditions.NO_OP,
+                            AtlasDbConstraintCheckingMode.NO_CONSTRAINT_CHECKING,
+                            null,
+                            TransactionReadSentinelBehavior.THROW_EXCEPTION,
+                            false,
+                            timestampValidationReadCache,
+                            getRangesExecutor,
+                            defaultGetRangesConcurrency,
+                            sweepQueueWriter,
+                            deleteExecutor,
+                            validateLocksOnReads,
+                            () -> TRANSACTION_CONFIG),
+                    counter);
+        }
+
+        @Override
+        protected Transaction createTransaction(
```

Comment:
how come there's no wrapping going on here? what's the point of the counter then even?

Reply:
Mistake, fixed.